addons/code-analysis/userscript.js

```diff
@@ -0,0 +1,195 @@
+  horizontalGradientContainer.appendChild(arrow);
+  horizontalGradientContainer.appendChild(createElementWithInnerHTML("ps-horizontal-description-motion", "span", "Motion and Appearance"));
+  horizontalGradientContainer.appendChild(createElementWithInnerHTML("ps-horizontal-description-logic", "span", "Logic and Algorithm"));
+  return horizontalGradientContainer;
+}
+
+function createCodeGenetic(addon, total, classifiedCodeMap) {
+  const horizontalSlider = createElement("ps-code-genetic", "div");
+  horizontalSlider.appendChild(createElement("ps-code-genetic-bg", "span"));
+  let baseLine = 46;
+  codeGeneticList.forEach((item) => {
+    if (total === 0) {
+      return;
+    }
+    const block = createElement(`ps-code-genetic-block ${item}`, "span");
+    const num = getterMap[item](classifiedCodeMap);
+    const width = num * 837 / total;
+    block.style.left = `${baseLine}px`;
+    block.style.width = `${width}px`;
+    horizontalSlider.appendChild(block);
+    baseLine += width;
+  });
+  horizontalSlider.appendChild(Object.assign(document.createElement("img"), {
+    className: "ps-edge01",
+    src: addon.self.dir + "/edge.svg"
+  }));
+  horizontalSlider.appendChild(Object.assign(document.createElement("img"), {
+    className: "ps-edge02",
+    src: addon.self.dir + "/edge.svg"
+  }));
+  return horizontalSlider;
+}
+
+function createCodeGeneticTable(total, classifiedCodeMap) {
+  const codeGeneticTable = createElement("ps-code-genetic-table", "div");
+  let sum = 0;
+  codeGeneticTableList.forEach((items, index) => {
+    items.forEach((item, nestedIndex) => {
+      codeGeneticTable.appendChild(createElementWithInnerHTML(`ps-code-genetic-item-${item}`, "span", `${item.replace("-", " ")}:`));
+      const codeSum = getterMap[item](classifiedCodeMap);
+      codeGeneticTable.appendChild(createElementWithInnerHTML(`ps-code-genetic-item row${index} col${nestedIndex}`, "span", codeSum));
+      sum += codeSum;
+    });
+  });
+  codeGeneticTable.appendChild(createElementWithInnerHTML("ps-code-genetic-item-extensions", "span", "Extensions:"));
+  codeGeneticTable.appendChild(createElementWithInnerHTML("ps-code-genetic-item row0 col3", "span", total - sum));
+  codeGeneticTable.appendChild(createElementWithInnerHTML("ps-code-genetic-footer", "span", "Code Genetic Map"));
+  return codeGeneticTable;
+}
+
+function createSliderBG(name, value) {
+  const psSliderBG = createElement(`ps-slider-bg ${name}`, "span");
+  psSliderBG.appendChild(createElementWithInnerHTML("ps-slider-title", "text", name.toUpperCase()));
+  psSliderBG.appendChild(createElementWithInnerHTML("ps-slider-content", "text", value));
+  return psSliderBG;
+}
+
+function createElement(className, label) {
+  const element = document.createElement(label);
+  element.className = className;
+  return element;
+}
+
+function createElementWithInnerHTML(className, label, innerHTML) {
+  const element = createElement(className, label);
+  element.innerHTML = innerHTML;
```

Comment:
https://scratchaddons.com/docs/develop/userscripts/best-practices/#avoid-using-innerhtml

Reply:
Haven't checked in detail, but it looks like all uses of this function would work too if using innerText. So you could just change it to that.